rosette_api.rb

```diff
@@ -23,164 +25,176 @@ class RosetteAPI
+
+    params = params.load_params
+
+    RequestBuilder.new(@user_key, @alternate_url + SENTIMENT_ENDPOINT, params)
+                  .send_post_request
   end
 
   def name_translation(params)
-    if params.instance_of? NameTranslationParameters
-      params = params.load_params
-      RequestBuilder.new(@user_key, @alternate_url + NAME_TRANSLATION_ENDPOINT, params).send_post_request
-    else
-      raise RosetteAPIError.new('badRequest', 'Expects a NameTranslationParameters type as an argument')
-    end
+    check_params params, 'Expects a NameTranslationParameters type as an argument', NameTranslationParameters
+
+    params = params.load_params
+
+    RequestBuilder.new(@user_key, @alternate_url + NAME_TRANSLATION_ENDPOINT, params)
+                  .send_post_request
   end
 
   def name_similarity(params)
-    if params.instance_of? NameSimilarityParameters
-      params = params.load_params
-      RequestBuilder.new(@user_key, @alternate_url + NAME_SIMILARITY_ENDPOINT, params).send_post_request
-    else
-      raise RosetteAPIError.new('badRequest', 'Expects a NameSimilarityParameters type as an argument')
-    end
+    check_params params, 'Expects a NameSimilarityParameters type as an argument', NameSimilarityParameters
+
+    params = params.load_params
+
+    RequestBuilder.new(@user_key, @alternate_url + NAME_SIMILARITY_ENDPOINT, params)
+                  .send_post_request
   end
 
   def get_tokens(params)
-    if params.instance_of? DocumentParameters
-      params = params.load_params
-      RequestBuilder.new(@user_key, @alternate_url + TOKENS_ENDPOINT, params).send_post_request
-    else
-      raise RosetteAPIError.new('badRequest', 'Expects a DocumentParameters type as an argument')
-    end
+    check_params params
+
+    params = params.load_params
+
+    RequestBuilder.new(@user_key, @alternate_url + TOKENS_ENDPOINT, params)
+                  .send_post_request
   end
 
   def get_sentences(params)
-    if params.instance_of? DocumentParameters
-      params = params.load_params
-      RequestBuilder.new(@user_key, @alternate_url + SENTENCES_ENDPOINT, params).send_post_request
-    else
-      raise RosetteAPIError.new('badRequest', 'Expects a DocumentParameters type as an argument')
-    end
+    check_params params
+
+    params = params.load_params
+
+    RequestBuilder.new(@user_key, @alternate_url + SENTENCES_ENDPOINT, params)
+                  .send_post_request
   end
 
   def info
-    RequestBuilder.new(@user_key, @alternate_url + INFO).send_get_request
+    RequestBuilder.new(@user_key, @alternate_url + INFO)
+                  .send_get_request
   end
 
   def ping
-    RequestBuilder.new(@user_key, @alternate_url + PING).send_get_request
+    RequestBuilder.new(@user_key, @alternate_url + PING)
+                  .send_get_request
   end
-end
\ No newline at end of file
+
+  private
+
+    def check_params(params, message = 'Expects a DocumentParameters type as an argument', type = DocumentParameters)
```

Comment:
I've seen a couple of variations on private methods. I personally prefer indenting them under the `private` keyword.